app/src/main/java/org/schabi/newpipe/DownloaderImpl.java

```diff
@@ -124,59 +124,43 @@ public long getContentLength(final String url) throws IOException {
     }
 
     @Override
-    public Response execute(@NonNull final Request request)
-            throws IOException, ReCaptchaException {
+    public Response execute(@NonNull final Request request) throws IOException, ReCaptchaException {
         final String httpMethod = request.httpMethod();
         final String url = request.url();
         final Map<String, List<String>> headers = request.headers();
         final byte[] dataToSend = request.dataToSend();
 
-        RequestBody requestBody = null;
-        if (dataToSend != null) {
-            requestBody = RequestBody.create(dataToSend);
-        }
+        final RequestBody requestBody = Optional.ofNullable(dataToSend)
+                .map(RequestBody::create)
+                .orElse(null);
 
         final okhttp3.Request.Builder requestBuilder = new okhttp3.Request.Builder()
-                .method(httpMethod, requestBody).url(url)
+                .method(httpMethod, requestBody)
+                .url(url)
                 .addHeader("User-Agent", USER_AGENT);
 
         final String cookies = getCookies(url);
         if (!cookies.isEmpty()) {
             requestBuilder.addHeader("Cookie", cookies);
         }
 
-        for (final Map.Entry<String, List<String>> pair : headers.entrySet()) {
-            final String headerName = pair.getKey();
-            final List<String> headerValueList = pair.getValue();
-
-            if (headerValueList.size() > 1) {
-                requestBuilder.removeHeader(headerName);
-                for (final String headerValue : headerValueList) {
-                    requestBuilder.addHeader(headerName, headerValue);
-                }
-            } else if (headerValueList.size() == 1) {
-                requestBuilder.header(headerName, headerValueList.get(0));
```

Comment:
Looking at the impl of `.header` it’s the same as `.removeHeader().addHeader()` so the change should be fine.